File: src/api/processes.ts
import { invoke } from "@tauri-apps/api/core";
import type { WowModificationSafetyState } from "../models/processes";

export async function getWowModificationSafetyState(): Promise<WowModificationSafetyState> {
  return invoke<WowModificationSafetyState>("get_wow_modification_safety_state");
}

File: src-tauri/src/commands/processes.rs
//! Commands that expose the centralized WoW process safety state.

use crate::processes::wow::{inspect_wow_processes, WowModificationSafetyState};

#[tauri::command]
pub fn get_wow_modification_safety_state() -> WowModificationSafetyState {
    inspect_wow_processes()
}
